### PR TITLE
[CoreML Backend] Update coreml runner to only profile model when prof…

### PR DESCRIPTION
**Pull request by cymbalrush**
The CoreML executor would profile the model by default, this causes an issue when the installed sdk < 14.4. Model execution fails as profiling is only available for >= 14.4 and on older sdk it returns an error.

This addresses the issue by fixing the default option and only profiling when `profiling_model` option is set.

### CODE REVIEW

**Comment by cymbalrush** [comment]
@guangy10 I am unable to add `ciflow/trunk` label, any suggestions?

**Comment by guangy10** [comment]
> @guangy10 I am unable to add `ciflow/trunk` label, any suggestions?

@cymbalrush I'm a bit surprised that you were not added to the repo as a contributor. Just added you in. Can you try it again?

**Comment by cymbalrush** [comment]
@guangy10 still the same issue

**Comment by guangy10** [comment]
> Can't add following labels to PR: ciflow/trunk. Please ping one of the reviewers for help.

@cymbalrush will add that tag to unblock. At meanwhile @huydhn can you take a look?

**Comment by huydhn** [comment]
We have tighten the security when it comes to running CI job recently, if you don't have write access to the repo, the CI will not be run by default.  You will need to ask the reviewer to approve CI run explicitly, i.e.

Screenshot 2024-03-21 at 18 48 56

Once this is approved, then attaching `ciflow/trunk` will be allow to start trunk jobs.  Note that updating the PR by pushing a new commit will invalidate the approval.

cc @guangy10 For regular contributors to ET that have a proven record, you could consider granting them write access to the repo, then they will not need to ask for CI approval.

**Comment by guangy10** [comment]
> We have tighten the security when it comes to running CI jobs recently, if you don't have write access to the repo, the CI will not be run by default. You will need to ask the reviewer to approve CI run explicitly, i.e.
> 

@huydhn Oh, this is sad as relying on pytorchbot to kick off trunk/periodic jobs will be meaningless. That doesn't sound a good move, especially for ET which is not in GH1st.

**Comment by guangy10** [comment]
@cymbalrush FYI, I granted you the write permission to unblock.

**Comment by cymbalrush** [comment]
Thanks @guangy10 @huydhn !

**Comment by cymbalrush** [comment]
I am seeing failures unrelated to the radar, the changes in the PR just touch the CoreML runner, the failure is happening at quant_fusion_pass.py

packages/executorch/exir/passes/__init__.py", line 45, in <module>
    from executorch.exir.passes.quant_fusion_pass import QuantFusionPass
  File "/Users/runner/work/_temp/conda_environment_8384223477/lib/python3.11/site-packages/executorch/exir/passes/quant_fusion_pass.py", line 14, in <module>
    from ._quant_patterns_and_replacements import get_quant_patterns_and_replacements
  File "/Users/runner/work/_temp/conda_environment_8384223477/lib/python3.11/site-packages/executorch/exir/passes/_quant_patterns_and_replacements.py", line 17, in <module>
    from torchao.quantization.quant_primitives import quantized_decomposed_lib
  File "/Users/runner/work/_temp/conda_environment_8384223477/lib/python3.11/site-packages/torchao/quantization/__init__.py", line 7, in <module>
    from .smoothquant import *  # noqa: F403
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/_temp/conda_environment_8384223477/lib/python3.11/site-packages/torchao/quantization/smoothquant.py", line 17, in <module>
    import torchao.quantization.quant_api as quant_api
  File "/Users/runner/work/_temp/conda_environment_8384223477/lib/python3.11/site-packages/torchao/quantization/quant_api.py", line 25, in <module>
    from .dynamic_quant import DynamicallyPerAxisQuantizedLinear
  File "/Users/runner/work/_temp/conda_environment_8384223477/lib/python3.11/site-packages/torchao/quantization/dynamic_quant.py", line 9, in <module>
    from .quant_primitives import (
  File "/Users/runner/work/_temp/conda_environment_8384223477/lib/python3.11/site-packages/torchao/quantization/quant_primitives.py", line 19, in <module>
    from torchao.kernel.intmm_triton import int_scaled_matmul
  File "/Users/runner/work/_temp/conda_environment_8384223477/lib/python3.11/site-packages/torchao/kernel/intmm_triton.py", line 6, in <module>
    import triton
